code_migration_v1.1_to_v1.2.md

```diff
@@ -0,0 +1,144 @@
+	task_manager.spawn_handle(),
+	client.clone(),
+	transaction_pool.clone(),
+	prometheus_registry.as_ref(),
+	telemetry.as_ref().map(|x| x.handle()),
+);
+let proposer_factory: PartnerChainsProposerFactory<_, _, McHashInherentDigest> =
+	PartnerChainsProposerFactory::new(basic_authorship_proposer_factory);
+```
+
+## pallet-partner-chains-session
+
+This pallet was moved from polkadot-sdk to partner-chains, please update your cargo files.
+Also add `pallet-session-runtime-stub` from partner-chains to your runtime dependencies.
+Add `pallet-session` from polkadot-sdk as well to runtime and node dependencies.
+
+## runtime code
+
+It is required to add `pallet-session` `Runtime` configuration to your `construct_runtime!` macro.
+What is more, currently it has to be stubbed implementation, because of how `pallet-partner-chains-session` is implemented and wired in.
+To obtain this stub implementation use:
+```rust
+pallet_session_runtime_stub::impl_pallet_session_config!(Runtime);
+```
+Then add: `PolkadotSession:pallet_session` in the `construct_runtime!` macro.
+
+## sc-cli
+
+`Runner` has been reverted to pairtytech version, this requires changes in the node code,
+because the vanilla polkadot-sdk version doesn't support async in a way our custom code did.
+
+Required change is in *service.rs* file, where the main chain follower has to be created in a blocking way:
+```rust
+let data_sources = task::block_in_place(|| {
+	config.tokio_handle.block_on(
+		crate::main_chain_follower::create_cached_main_chain_follower_data_sources(
+			mc_follower_metrics.clone(),
+		),
+	)
+})?;
+```
+
+`pub fn new_partial` is not async anymore, please follow the change in your implementation.
+
+Please remove `async move` and `.await` in the `run` function of your node in places where compiler complains.
+For example replace:
+```rust
+Some(Subcommand::CheckBlock(cmd)) => {
+	let runner = cli.create_runner(cmd)?;
+	runner.async_run(|config| async move {
+		let PartialComponents { client, task_manager, import_queue, .. } =
+			service::new_partial(&config).await?;
+		Ok((cmd.run(client, import_queue), task_manager))
+	})
+},
+```
+with:
+```rust
+Some(Subcommand::CheckBlock(cmd)) => {
+	let runner = cli.create_runner(cmd)?;
+	runner.async_run(|config| {
+		let PartialComponents { client, task_manager, import_queue, .. } =
+			service::new_partial(&config)?;
+		Ok((cmd.run(client, import_queue), task_manager))
+	})
+},
+```
+
+## chain-init feature
+
+Code that was present in `node` crate related to `USE_CHAIN_INIT` feature has been removed.
+The only supported way to obtain the chain-spec is through partner-chains-cli.
+Please cleanup your node if it contained copy-pasted code for it.
+
+# Observation of native token illiquid supply address
+
+This feature is not complete, please do not follow changes in partner-chains reference runtime and node.
+Please do not wire it in your node code nor use the "native-token" feature in the main chain follower.
```

Comment:
If this feature is supported in v1.2 it has separate ticket for migration guide: https://input-output.atlassian.net/browse/ETCM-8366